### PR TITLE
feat: add `azcopy` to Linux provisioning

### PR DESCRIPTION
**Pull request by lemeurherve**
This PR adds https://github.com/Azure/azure-storage-azcopy to Linux provisioning so we can use it in https://github.com/jenkins-infra/crawler/ job to synchronize the Update Center File Share that will be used by updates.jenkins.io.

Ref:
- https://github.com/jenkins-infra/helpdesk/issues/2649#issuecomment-1807816044

### CODE REVIEW

File: goss/goss-linux.yaml [hunk 1]
```diff
@@ -14,6 +14,11 @@ command:
     exit-status: 0
     stdout:
       - 2.53.1
+  azcopy:
```
Comment (lemeurherve) [inline goss/goss-linux.yaml:17]:
```suggestion
  azcopy: # not tracked by updatecli
```

File: provisioning/ubuntu-provision.sh [hunk 2]
```diff
@@ -403,6 +403,15 @@ function install_azurecli() {
   apt-get install --yes --no-install-recommends azure-cli="${AZURECLI_VERSION}*"
 }
 
+## Ensure that azcopy is installed
+function install_azcopy() {
```
Comment (dduportal) [inline provisioning/ubuntu-provision.sh:407]: Don't forget to call the function 😅

**Review by dduportal — CHANGES_REQUESTED** [review]
Almost there: you have to call the function

File: provisioning/ubuntu-provision.sh (hunk 2 above)
Comment (dduportal) [reply provisioning/ubuntu-provision.sh:407]: Fixup commit pushed (https://github.com/jenkins-infra/packer-images/pull/898/commits/3f832ab738b637989ab74b5648af61362dc56ed2)

**Review by dduportal — APPROVED** [review]

**Pull request merged**